gk/main.c

```diff
@@ -41,6 +329,17 @@ gk_proc(void *arg)
 
 	rte_atomic32_inc(&gk_conf->ref_cnt);
 
+	/* Setup the flow entry table using hash table. */
+	ret = setup_gk_instance(block_idx, gk_conf);
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER, 
+			"gk: failed to setup gk instances for GK block at %u!\n",
+			lcore);
+		exiting = true;
```

Comment:
Should this setup happen in the `for` loop in `gk_run()` instead?

If the initialization fails in `gk_run()`, it returns an error which then halts the rest of the configuration. However, if `gk_proc()` needs to quit, all it can do is set `exiting` to `true`, but other blocks will continue with their initialization, allocation, etc.

In other words, I think it would be better if all fail-able setup is done in the `run_xxx()` functions, which can fail gracefully. This also keeps the number of writers of `exiting` to a minimum. Any thoughts?